src/backend/marsha/core/api/video.py

```diff
@@ -719,6 +733,98 @@ def push_attendance(self, request, pk=None):
+            )
+        return Response(
+            {
+                "display_username": liveregistration.display_username,
+                "username": liveregistration.username,
+            }
+        )
+
+    @action(detail=False, methods=["patch"], url_path="set_display_username")
+    # pylint: disable=unused-argument
+    def set_display_username(self, request, pk=None):
+        """View handling setting display_username. Create or get registration."""
+        serializer = serializers.LiveRegistrationDisplayUsernameSerializer(
+            data=request.data
+        )
+        if not serializer.is_valid() or not request.data.get("display_username"):
+            return Response({"detail": "Invalid request."}, status=400)
+
+        user = self.request.user
+        video = get_object_or_404(Video, pk=user.id)
+
+        try:
+            update_fields = {
+                "display_username": serializer.validated_data["display_username"],
+            }
+            if self.is_lti_token():
+                token_user = user.token.payload.get("user")
+                consumer_site = get_object_or_404(
+                    ConsumerSite, pk=user.token.payload["consumer_site"]
+                )
+                # Update email only if it's defined in the token user
+                if token_user.get("email"):
+                    update_fields = update_fields | {"email": token_user.get("email")}
+
+                # Update username only it's defined in the token user
+                if token_user.get("username"):
+                    update_fields = update_fields | {
+                        "username": token_user.get("username")
+                    }
+
+                liveregistration, _ = LiveRegistration.objects.update_or_create(
+                    consumer_site=consumer_site,
+                    lti_id=user.token.payload.get("context_id"),
+                    lti_user_id=token_user.get("id"),
+                    video=video,
+                    defaults=update_fields,
+                )
+            else:
+                liveregistration, _ = LiveRegistration.objects.update_or_create(
+                    anonymous_id=serializer.validated_data["anonymous_id"],
+                    video=video,
+                    defaults=update_fields,
+                )
+            return Response(
+                {
+                    "display_username": liveregistration.display_username,
+                    "username": liveregistration.username,
+                }
+            )
+        except IntegrityError as error:
+            if "liveregistration_unique_video_display_username" in error.args[0]:
+                return Response(
+                    {"display_username": "User with that username already exists!"},
+                    status=status.HTTP_409_CONFLICT,
+                )
+
+            return Response(
+                {"display_username": error}, status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
I think we could raise other errors like this.
```suggestion
            raise error
```
With this, I think we don't have to mock anything to raise IntegrityError, as we only handle `liveregistration_unique_video_display_username` error. WDYT ?

Reply:
if you say so lol! it's good to me :-)